java/core/src/java/org/apache/orc/impl/ColumnStatisticsImpl.java

```diff
@@ -993,8 +993,7 @@ public void updateBinary(byte[] bytes, int offset, int length,
 
     @Override
     public void merge(ColumnStatisticsImpl other) {
-      if (other instanceof BinaryColumnStatistics) {
-        BinaryStatisticsImpl bin = (BinaryStatisticsImpl) other;
+      if (other instanceof BinaryStatisticsImpl bin) {
         sum += bin.sum;
       } else {
```

Comment:
BTW, to be clear, this PR changes like the following so that `else` statement need to handle more silently in the AS-IS form. I believe it's wrong. Please handle the type incompatible case before reaching out  this `else` statement. In other words, I guess we may need another `else` statement before this `else`.
```
- if (other instanceof BinaryColumnStatistics) {
+ if (other instanceof BinaryStatisticsImpl bin) {
```

Reply:
How about just removing the if judgment in else?
For example, removed `if (isStatsExists() && sum != 0) {`

```java
      if (other instanceof BinaryStatisticsImpl bin) {
        sum += bin.sum;
      } else {
        throw new IllegalArgumentException("Incompatible merging of binary column statistics");
      }
```

Reply:
Ur, could you elaborate your theoretic background of removal? How do you know that is not a regression and safe?